src/main/java/org/testng/IFactoryMethod.java

```diff
@@ -0,0 +1,41 @@
+package org.testng;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+/**
+ * Represents a Factory method that was involved for producing test class instances.
+ */
+public interface IFactoryMethod {
+
+    /**
+     * @return - The class to which the factory method belongs to.
+     */
+    Class<?> getDeclaringClass();
+
+    /**
+     * @return - The name of the factory method. Could either be the constructor's name or the static factory
+     * method's name.
+     */
+    String getName();
+
+    /**
+     * @return - An array of parameters that are part of the factory method.
+     */
+    Class[] getParameterTypes();
+
+    /**
+     * @return - The <code>@Factory</code> annotated method.
+     */
+    Method getMethod();
```

Comment:
replace get method/constructor by get `IConstructorOrMethod`

Reply:
`IConstructorOrMethod` ? That doesn't exist as of now. Can you please tell me what you are hinting at here ?

Reply:
It doesn't exist for the moment but as we don't want to expose the implementation, we may add a new interface

Reply:
So are you suggesting that we get rid of the `IFactoryMethod` interface and instead replace it with a `IConstructorOrMethod` interface ?

Reply:
For me, `ConstructorOrMethod` is just a way to manage targets we want to run but we don't care if it is a constructor or a method. In fact, we should not use `Method` or `Constructor` from the TestNG implementation.
`IConstructorOrMethod` is just a way to not expose `ConstructorOrMethod` directly.
`IFactoryMethod` is just the modelization of the factory method and from it, we need the target (`IConstructorOrMethod`), its annotation (directly or by methods), its optional data provider, its optional params, ...